### PR TITLE
Save random_indexes for the RB

### PR DESCRIPTION
**Pull request by Jacfomg**
>[!CAUTION]
> It requieres https://github.com/qiboteam/qibolab/pull/864 to fix RB completely.

This will save the random indexes in the data.json from the reports. And fix the RB random number generation.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.
- [ ] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [ ] Qibo: `master`
    - [ ] Qibolab: `main`
    - [ ] Qibolab_platforms_qrc: `main`

### CODE REVIEW

**Comment by Jacfomg** [comment]
@Edoardo-Pedicillo please take a look into the saving, but mostly into the class I made to simplify and make things properly random

**Review by Edoardo-Pedicillo — APPROVED** [review]
Thanks

File: src/qibocal/protocols/characterization/randomized_benchmarking/standard_rb.py [hunk 1]
```diff
@@ -84,8 +84,8 @@ class RBData(Data):
     """Number of iterations for each depth."""
     data: dict[QubitId, npt.NDArray[RBType]] = field(default_factory=dict)
     """Raw data acquired."""
-    circuits: Circuit = None
-    """Circuits executed."""
+    circuits: dict[QubitId, int] = field(default_factory=dict)
```
Comment (andrea-pasquale) [inline src/qibocal/protocols/characterization/randomized_benchmarking/standard_rb.py:87]: Wrong type hint

File: src/qibocal/protocols/characterization/randomized_benchmarking/standard_rb.py [hunk 2]
```diff
@@ -117,16 +117,29 @@ def __contains__(self, qubit: QubitId):
         return True
 
 
-def layer_gen(targets, seed):
-    """Returns a circuit with a random single-qubit clifford unitary."""
-    return random_clifford(targets, seed)
+class RB_Generator:
+
```
Comment (andrea-pasquale) [inline src/qibocal/protocols/characterization/randomized_benchmarking/standard_rb.py:121]: Docstring

File: src/qibocal/protocols/characterization/randomized_benchmarking/utils.py [hunk 3]
```diff
@@ -56,34 +55,10 @@ def random_clifford(qubits, seed=None):
...
-
-    if isinstance(qubits, int):
-        qubits = list(range(qubits))
-
-    random_indexes = local_state.integers(0, len(SINGLE_QUBIT_CLIFFORDS), len(qubits))
-    clifford_gates = [
-        SINGLE_QUBIT_CLIFFORDS[p](q) for p, q in zip(random_indexes, qubits)
-    ]
+    random_index = int(random_index_gen(SINGLE_QUBIT_CLIFFORDS))
+    clifford_gates = SINGLE_QUBIT_CLIFFORDS[random_index](0)
```
Comment (andrea-pasquale) [inline src/qibocal/protocols/characterization/randomized_benchmarking/utils.py:59]: 👍🏻

File: tests/test_randomized_benchmarking.py [hunk 4]
```diff
@@ -143,15 +146,8 @@ def test_model(noise_model, num_keys=1):
...
 def test_random_clifford(qubits, seed):
-    with pytest.raises(TypeError):
-        q = "1"
-        random_clifford(q)
-    with pytest.raises(ValueError):
-        q = -1
-        random_clifford(q)
-    with pytest.raises(ValueError):
-        q = [0, 1, -3]
-        random_clifford(q)
```
Comment (andrea-pasquale) [inline tests/test_randomized_benchmarking.py:154]: 👍🏻

File: tests/test_randomized_benchmarking.py [hunk 5]
```diff
@@ -164,9 +160,15 @@ def test_random_clifford(qubits, seed):
...
 
-    result = result_single if (isinstance(qubits, int) and qubits == 1) else result_two
+    result = result_single if isinstance(qubits, int) else result_two
+
+    if isinstance(qubits, int):
+        qubits = [qubits]
+    gates = []
+    for qubit in qubits:
+        gate, index = random_clifford(rb_gen.random_index)
+        gates.append(gate)
```
Comment (andrea-pasquale) [inline tests/test_randomized_benchmarking.py:170]: A bit ugly but fine

**Review by andrea-pasquale — APPROVED** [review]

File: tests/test_randomized_benchmarking.py (hunk 5 above)
Comment (Jacfomg) [reply tests/test_randomized_benchmarking.py:170]: Any suggestion ?

**Pull request merged**